submission/pipelines/generate_pipelines.py

```diff
@@ -400,29 +402,71 @@ def remove_digests(
     return pipeline_json_structure
 
 
+def update_digest(
+    pipeline_json_structure, filename=None
+):
+    """
+    This function updates the pipeline's digests and version numbers
+
+    Parameters
+    ----------
+    pipeline_json_structure: the pipeline in JSON form (WITHOUT) digests.  This or the `filename` parameter is mandatory
+    filename: the filename of the pipeline json, so we can read it in
+
+    :return a pipeline with updated digests
+    """
+    if pipeline_json_structure is None and filename is None:
+        raise Exception
+    elif pipeline_json_structure is None:
+        with open(filename, "r") as file:
+            # NOTE: must be a pipeline with no digests, or recent digests
+            # NOTE: reading this in as straight JSON doesn't work so we have to use the pipeline_module
+            pipeline_to_run = pipeline_module.Pipeline.from_json(string_or_file=file).to_json_structure()
+    else:
+        pipeline_to_run = pipeline_module.Pipeline.from_json(json.dumps(pipeline_json_structure)).to_json_structure()
+
+    for step in pipeline_to_run['steps']:
+        # if not updated, check and update
+        primitive = pipeline_module.PrimitiveStep(
+            primitive=d3m_index.get_primitive(
+                step["primitive"]["python_path"]
+            )
+        )
+        check_step = primitive.to_json_structure()
+        # lets verify that both are updated
+        assert check_step["primitive"]["version"] == step["primitive"]["version"], "Updating version failed"
+        assert check_step["primitive"]["digest"] == step["primitive"]["digest"], "Updating digest failed"
+
+    return pipeline_to_run
+
+
+# get directory ready
+byu_dir = get_new_d3m_path()
+clear_directory(byu_dir)
+
 for task_type in ['classification', 'regression']:
+    # generate and update imputer
     pipeline = generate_imputer_pipeline(task_type)
     pipeline_json_structure = pipeline.to_json_structure()
-    remove_digests(
-        pipeline_json_structure, exclude_primitives={
-            RandomSamplingImputer.metadata.query()['id']
-        }
-    )
-    pipeline_path = './pipelines/random_sampling_imputer/{}.json'.format(
-        pipeline_json_structure['id']
-    )
-    json.dump(pipeline_json_structure, open(pipeline_path, 'w'), indent=4)
-    os.chmod(pipeline_path, 0o777)
-
+    pipeline_json_structure = remove_digests(pipeline_json_structure, exclude_primitives={
+                                             RandomSamplingImputer.metadata.query()['id']})
+    pipeline_json_structure = update_digest(pipeline_json_structure)
+    # place in submodule
+    imputer_path, imputer_version = extract_byu_info(pipeline_json_structure)
```

Comment:
you can get this information from the primitives themselves. no need for an extra function